Subject: [PATCH] poor_logic
---
Index: hash-service/src/main/java/com/otus/javaadvanced/services/ExceptionUtils.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/hash-service/src/main/java/com/otus/javaadvanced/services/ExceptionUtils.java b/hash-service/src/main/java/com/otus/javaadvanced/services/ExceptionUtils.java
--- a/hash-service/src/main/java/com/otus/javaadvanced/services/ExceptionUtils.java	(revision 5e6efa571cc18f8438c376d9b68df2e5bb6cee01)
+++ b/hash-service/src/main/java/com/otus/javaadvanced/services/ExceptionUtils.java	(date 1739808845562)
@@ -15,7 +15,10 @@
     public static long countException(Exception e) {
         lock.lock();
         try {
-            return ++exceptionCount;
+            int load = 1;
+            for (int i = 0; i < 1_000_000; i++) load=load*load; // poor logic
+
+            return ++exceptionCount * load;
         } finally {
             lock.unlock();
         }
Index: hash-service/src/main/java/com/otus/javaadvanced/services/EncodeService.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/hash-service/src/main/java/com/otus/javaadvanced/services/EncodeService.java b/hash-service/src/main/java/com/otus/javaadvanced/services/EncodeService.java
--- a/hash-service/src/main/java/com/otus/javaadvanced/services/EncodeService.java	(revision 5e6efa571cc18f8438c376d9b68df2e5bb6cee01)
+++ b/hash-service/src/main/java/com/otus/javaadvanced/services/EncodeService.java	(date 1739808845557)
@@ -8,10 +8,10 @@
 @RequiredArgsConstructor
 public class EncodeService {
 
-    private final Encoder encoder;
+    private Encoder encoder;
 
     // for emulation of large stacks
-    private static final int STACK_DEPTH = 0;
+    private static final int STACK_DEPTH = 50;
 
     public byte[] encode(byte[] hash) {
         byte[] safe = Arrays.copyOf(hash, hash.length);
Index: hash-service/src/main/resources/application.yml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/hash-service/src/main/resources/application.yml b/hash-service/src/main/resources/application.yml
--- a/hash-service/src/main/resources/application.yml	(revision 5e6efa571cc18f8438c376d9b68df2e5bb6cee01)
+++ b/hash-service/src/main/resources/application.yml	(date 1739808700454)
@@ -6,8 +6,8 @@
   port: 8080
   tomcat:
     threads:
-      min-spare: 10
-      max: 200
+      min-spare: 1
+      max: 3
   shutdown: graceful
 
 management:
Index: hash-service/src/main/java/com/otus/javaadvanced/config/Config.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/hash-service/src/main/java/com/otus/javaadvanced/config/Config.java b/hash-service/src/main/java/com/otus/javaadvanced/config/Config.java
--- a/hash-service/src/main/java/com/otus/javaadvanced/config/Config.java	(revision 5e6efa571cc18f8438c376d9b68df2e5bb6cee01)
+++ b/hash-service/src/main/java/com/otus/javaadvanced/config/Config.java	(date 1739808845551)
@@ -18,13 +18,8 @@
     }
 
     @Bean
-    public EncodeService encodeService(Encoder encoder) {
-        return new EncodeService(encoder);
-    }
-
-    @Bean
-    public Encoder encoder() {
-        return (src) -> Base64.getEncoder().encode(src);
+    public EncodeService encodeService() {
+        return new EncodeService();
     }
 
     @Bean
